File: app/src/main/java/com/lumiyaviewer/lumiya/voice/common/messages/VoiceRejectCall.java
package com.lumiyaviewer.lumiya.voice.common.messages;

import android.os.Bundle;
import com.lumiyaviewer.lumiya.voice.common.VoicePluginMessage;
import com.lumiyaviewer.lumiya.voice.common.model.VoiceChannelInfo;
import javax.annotation.Nonnull;

public class VoiceRejectCall implements VoicePluginMessage {
    @Nonnull
    public final String sessionHandle;
    public final VoiceChannelInfo voiceChannelInfo;

    public VoiceRejectCall(Bundle bundle) {
        this.sessionHandle = bundle.getString("sessionHandle");
        this.voiceChannelInfo = new VoiceChannelInfo(bundle.getBundle("voiceChannelInfo"));
    }

    public VoiceRejectCall(@Nonnull String str, VoiceChannelInfo voiceChannelInfo2) {
        this.sessionHandle = str;
        this.voiceChannelInfo = voiceChannelInfo2;
    }

    public Bundle toBundle() {
        Bundle bundle = new Bundle();
        bundle.putString("sessionHandle", this.sessionHandle);
        bundle.putBundle("voiceChannelInfo", this.voiceChannelInfo.toBundle());
        return bundle;
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/voice/common/messages/VoiceRejectCall.kt
package com.lumiyaviewer.lumiya.voice.common.messages

import android.os.Bundle
import com.lumiyaviewer.lumiya.voice.common.VoicePluginMessage
import com.lumiyaviewer.lumiya.voice.common.model.VoiceChannelInfo
import javax.annotation.Nonnull

class VoiceRejectCall(
    val sessionHandle: String?,
    val voiceChannelInfo: VoiceChannelInfo
) : VoicePluginMessage {
    
    constructor(bundle: Bundle) : this(
        bundle.getString("sessionHandle"),
        VoiceChannelInfo(bundle.getBundle("voiceChannelInfo")!!)
    )

    override fun toBundle(): Bundle {
        val bundle = Bundle()
        bundle.putString("sessionHandle", sessionHandle)
        bundle.putBundle("voiceChannelInfo", voiceChannelInfo.toBundle())
        return bundle
    }
}